mobile/src/main/java/org/openhab/habdroid/core/OnUpdateBroadcastReceiver.java

```diff
@@ -0,0 +1,49 @@
+package org.openhab.habdroid.core;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.preference.PreferenceManager;
+import android.util.Log;
+
+import org.openhab.habdroid.ui.OpenHABMainActivity;
+import org.openhab.habdroid.util.Constants;
+
+public class OnUpdateBroadcastReceiver extends BroadcastReceiver {
+    private static final String TAG = OpenHABMainActivity.class.getSimpleName();
+
+    private static final String UPDATE_LOCAL_CREDENTIALS = "1.8.0.8";
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        if (getVersionNumber(context.getPackageManager(), context.getPackageName())
+                .equals(UPDATE_LOCAL_CREDENTIALS)) {
```

Comment:
This won't work if a user skips a version when updating. Preference versioning (storing the old version along with the data) likely will work better here.

Reply:
Hmm, you're right. Just to confirm that I understand correctly what you mean:
- There should be a preference, which saves the last used version (which needs to be updated with every onCreate of OpenHABMainActivity?)
- Here we check, if the last version is != the current version, and if so, update the username/password?

That would also require, that we remove this code snippet after we released the version, right?

Reply:
Ok, then probably I haven't understood the idea completely :) If we leave the code here over more versions, how we should decide, if a migration is really needed? Sorry, if the solution is pretty simple (which I assume), I really just don't get it :]

Reply:
Why? When installing an app for the first time, no conversion would be necessary as there's no existing data. When an installed app is updated, we'll get the broadcast.
And for deciding whether migration is needed, we'd read the current version number from prefs and decide based on the delta of the version numbers compared to the latest one supported by the new app version.
A challenge is indeed writing the version initially; we could probably make the settings activity write the latest app supported version unconditionally. On updates, we'll get the broadcast before the settings activity will run again.

Reply:
I probably "found" a way without needing to save the version in a shared pref at all, please take a look at my next commit, I would appreciate your feedback :]